File: solutions/19MajorityElement/java/MajorityElement.java
import java.util.HashMap;
import java.util.Map;

public class MajorityElement {
  public int majorityElement(int[] nums) {
    HashMap<Integer, Integer> numDict = new HashMap<>();
    for (int num : nums) {
      numDict.put(num, numDict.getOrDefault(num, 0) + 1);
    }

    // int maxCount = Integer.MIN_VALUE;
    // int maxElement = -1;

    // for (int num : numDict.keySet()) {
    // if (numDict.get(num) > maxCount) {
    // maxCount = numDict.get(num);
    // maxElement = num;
    // }
    // }
    for (Map.Entry<Integer, Integer> entry : numDict.entrySet()) {
      if (entry.getValue() > nums.length / 2) {
        return entry.getKey();
      }
    }
    // return maxElement;
    return 0;
  }
}

File: solutions/19MajorityElement/python/MajorityElement.py
class Solution(object):
    def majorityElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        numDict = dict()
        for num in nums:
            if num in numDict:
                numDict[num] += 1
            else:
                numDict[num] = 1

        max, max_index = float("-inf"), -1
        for num in numDict:
            if numDict[num] > max:
                max = numDict[num]
                max_index = num

        return max_index
